### PR TITLE
[Common API, Excel, OneNote, Word] Updating samples to follow patterns from Script Lab snippets

### PR DESCRIPTION
**Pull request by AlexJerabek**
**Reviewers**: Please note that only the **\*-snippets.yaml** need review. The others are autogenerated based on them.

This work is based on general customer feedback about samples. First, there have been requests to upgrade the samples to "modern" JS/TS. Second, we've observed confusion about the ref docs labelling API signatures as "TypeScript" while there are samples in "JavaScript". Finally, there have been cases of people using these older samples and encountering issues.

This PR changes the snippets present here in the following ways:
- `.then()` statements are (mostly) removed, in favor of `await` statements with `content.sync()`.
- `catch()` blocks were (mostly) removed to have a similar look to the Script Lab based snippets.
- Samples were compiled using the current TypeScript definitions and errors were fixed.
- A few samples were removed because they either:
  - Were redundant *or*
  - Didn't work and had no obvious path to fix. 

The Common API snippets were not evaluated as part of this. This is partially because there's merit in the 2013 samples working with ES5 JS and partially because of their relative priority. Visio snippets were not also addressed.

Outlook samples were not evaluated, as they use a callback structure and wouldn't have the same pattern of changes.

This PR addressed the occasional style issue. If it seems important, we can integrate that into this PR as well.

### CODE REVIEW

**Comment by AlexJerabek** [comment]
This PR is not high priority, but at some point could I please get reviews from @alison-mk on the Excel stuff and @ElizabethSamuel-MSFT on the other pieces?

File: docs/code-snippets/onenote-snippets.yaml [hunk 1]
```diff
@@ -985,45 +517,35 @@ OneNote.Page#copyToSection:member(1):
...
-    .catch(function(error) {
-        console.log("Error: " + error);
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
+
+        var section = notebook.sections.items[0];
+        
+        // copy page to the section.
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // Copy page to the section.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 2]
```diff
@@ -1776,40 +982,30 @@ OneNote.Section#insertSectionAsSibling:member(2):
...
-                // OneNote REST API according to the OneNote Development Blog.
-                // https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/
-            });
+        await context.sync();
+        console.log("The REST API ID is " + restApiId.value);
+        // Note that the REST API ID isn't all you need to interact with the OneNote REST API. 
+        // This is only required for SharePoint notebooks. baseUrl will be null for OneDrive notebooks.
+        // For SharePoint notebooks, the notebook baseUrl should be used to talk to the 
+        // OneNote REST API according to the OneNote Development Blog.
+        // https://blogs.msdn.microsoft.com/onenotedev/2015/06/11/and-sharepoint-makes-three/
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // https://docs.microsoft.com/archive/blogs/onenotedev/and-sharepoint-makes-three
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 3]
```diff
@@ -2166,510 +1174,310 @@ OneNote.SectionGroupCollection#load:member(2):
...
-                var paragraph = paragraphs.items[i];
-                if (paragraph.type == "Table") {
-                    var table = paragraph.table;
-                    var cell = table.getCell(2 /*Row Index*/, 3 /*Column Index*/);
-                }
+        await context.sync();
+
+        var paragraphs = outline.paragraphs;
+        
+        // for each table, get a cell in the second row and third column.
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // For each table, get a cell in the second row and third column.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 4]
```diff
@@ -2166,510 +1174,310 @@ OneNote.SectionGroupCollection#load:member(2):
...
-                var paragraph = paragraphs.items[i];
-                if (paragraph.type == "Table") {
-                    var table = paragraph.table;
-                    var row = table.insertRow(2, ["cell0", "cell1"]);
-                }
+        await context.sync()
+
+        var paragraphs = outline.paragraphs;
+        
+        // for each table, insert a row at index two.
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // For each table, insert a row at index two.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 5]
```diff
@@ -2166,510 +1174,310 @@ OneNote.SectionGroupCollection#load:member(2):
...
-                if (paragraph.type == "Table") {
-                    var table = paragraph.table;
-                    var cell = table.getCell(1 /*Row Index*/, 2 /*Column Index*/);
-                    appendedRichText = cell.appendRichText("Hello");
-                }
+        await context.sync();
+
+        var paragraphs = outline.paragraphs;
+        
+        // for each table, get a table cell at row one and column two and add "Hello".
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // For each table, get a table cell at row one and column two and add "Hello".
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 6]
```diff
@@ -2166,510 +1174,310 @@ OneNote.SectionGroupCollection#load:member(2):
...
-                        for (var i = 0; i < rows.items.length; i++) {
-                            console.log("Row " + i + " Parent Table Id: " + rows.items[i].parentTable.id);
-                            var cells = rows.items[i].cells;
-                            for (var j = 0 ; j < cells.items.length; j++) {
-                                console.log("Row " + i + " Cell " + j + " Id: " + cells.items[j].id);
-                            }
-                        }
-                        return ctx.sync();
-                    });
+        // for each table, get table rows.
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // For each table, get table rows.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 7]
```diff
@@ -2166,510 +1174,310 @@ OneNote.SectionGroupCollection#load:member(2):
...
+            if (paragraph.type == "Table") {
+                var table = paragraph.table;
+                
+                // Queue a command to load table.rows.
+                context.load(table, "rows");
+                await context.sync();
+
+                var rows = table.rows;
+                
+                // for each table row, log cell count and row index.
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
                // For each table row, log cell count and row index.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 8]
```diff
@@ -1314,25 +702,17 @@ OneNote.Paragraph#delete:member(1):
...
-    })
-    .catch(function(error) {
-        console.log("Error: " + error);
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        // Queue a command to delete the first paragraph                 
+        firstParagraph.delete();
+        
+        // Run the command to delete it
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // Queue a command to delete the first paragraph                 .
        firstParagraph.delete();
        
        // Run the command to delete it.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 9]
```diff
@@ -1347,26 +727,18 @@ OneNote.Paragraph#insertHtmlAsSibling:member(1):
...
-                // Run the command to run inserts
-                return context.sync();
-            });
-    ))
-    .catch(function(error) {
-        console.log("Error: " + error);
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        // Queue commands to insert before and after the first paragraph
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
        // Queue commands to insert before and after the first paragraph.
```

File: docs/code-snippets/onenote-snippets.yaml [hunk 10]
```diff
@@ -2166,510 +1174,310 @@ OneNote.SectionGroupCollection#load:member(2):
...
+        // For each table, get table rows.
+        for (var i = 0; i < paragraphs.items.length; i++) {
+            var paragraph = paragraphs.items[i];
+            if (paragraph.type == "Table") {
+                var table = paragraph.table;
+                
+                // Queue a command to load table.rows.
+                context.load(table, "rows");
+                
+                // Run the queued commands
```
Comment (ElizabethSamuel-MSFT) [inline docs/code-snippets/onenote-snippets.yaml]:
```suggestion
                // Run the queued commands.
```

**Review by ElizabethSamuel-MSFT — APPROVED** [review]
Changes make sense. Just had a few more clean-up suggestions.

File: docs/code-snippets/excel-snippets.yaml [hunk 11]
```diff
@@ -361,397 +229,249 @@ Excel.ChartCollection#add:member(1):
...
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
     });
 Excel.ChartGridlines#load:member(2):
   - |-
-    // Set to show major gridlines on valueAxis of Chart1
-    Excel.run(function (ctx) { 
-        var chart = ctx.workbook.worksheets.getItem("Sheet1").charts.getItem("Chart1");    
+    // Set to show major gridlines on valueAxis of Chart1.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Set the value axis of Chart1 to show the major gridlines.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 12]
```diff
@@ -361,397 +229,249 @@ Excel.ChartCollection#add:member(1):
...
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
+
+        console.log(legend.position);
     });
 Excel.ChartLineFormat#clear:member(1):
   - |-
     // Set to show legend of Chart1 and make it on top of the chart.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]: I don't think this comment describes this sample. 
```suggestion
    // Clear the format of the major gridlines on Chart1. 
```

File: docs/code-snippets/excel-snippets.yaml [hunk 13]
```diff
@@ -361,397 +229,249 @@ Excel.ChartCollection#add:member(1):
...
-            }
-        });
-    }).catch(function(error) {
-        console.log("Error: " + error);
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
-    });
   - |-
     // Get the number of chart series in collection.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Get the number of chart series in the collection.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 14]
```diff
@@ -361,397 +229,249 @@ Excel.ChartCollection#add:member(1):
...
-                console.log(title.text);
-        }).catch(function(error) {
-            console.log("Error: " + error);
-            if (error instanceof OfficeExtension.Error) {
-                console.log("Debug info: " + JSON.stringify(error.debugInfo));
-            }
-        });
-    });
   - |-
     // Set the text of Chart Title to "My Chart" and Make it show on top of the chart without overlaying.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]: If overlay=true then it seems like this comment is wrong. 
```suggestion
    // Set the text of the chart title to "My Chart" and display it as an overlay on the chart.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 15]
```diff
@@ -773,370 +493,257 @@ Excel.FilterOn:enum:
     });
 Excel.NamedItem#getRange:member(1):
   - |-
-    // Returns the Range object that is associated with the name. 
+    // Returns the Range object that is associated with the name.
     // null if the name is not of the type Range.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Returns `null` if the name is not of type Range.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 16]
```diff
@@ -773,370 +493,257 @@ Excel.FilterOn:enum:
...
+
+        for (var i = 0; i < nameditems.items.length; i++) {
+            console.log(nameditems.items[i].name);
         }
     });
 Excel.Range#clear:member(1):
   - |-
-    // Below example clears format and contents of the range. 
-    Excel.run(function (ctx) { 
+    // Below example clears format and contents of the range.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Clear the format and contents of the range.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 17]
```diff
@@ -773,370 +493,257 @@ Excel.FilterOn:enum:
...
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
+        
+        console.log(range.address); // prints Sheet1!A2:F2
     });
 Excel.Range#numberFormat:member:
   - |-
     // The example below sets number-format, values and formulas on a grid that contains 2x3 grid.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Set number format, values, and formulas on a grid that contains 2x3 grid.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 18]
```diff
@@ -1156,1010 +763,611 @@ Excel.Range#untrack:member(1):
 Excel.Range#load:member(2):
   - |-
     // Below example uses range address to get the range object.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Use the range address to get the range object.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 19]
```diff
@@ -1156,1010 +763,611 @@ Excel.Range#untrack:member(1):
...
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
+        
+        console.log(range.cellCount);
     });
 Excel.RangeBorder#style:member:
   - |-
     // The example below adds grid border around the range.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Add grid borders around the range.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 20]
```diff
@@ -1156,1010 +763,611 @@ Excel.Range#untrack:member(1):
...
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
+        console.log(rangeFont.name);
     });
 Excel.RangeFormat#load:member(2):
   - |-
-    // Below example selects all of the Range's format properties. 
-    Excel.run(function (ctx) { 
+    // Below example selects all of the Range's format properties.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Select all of the range's format properties.
```

File: docs/code-snippets/excel-snippets.yaml [hunk 21]
```diff
@@ -1156,1010 +763,611 @@ Excel.Range#untrack:member(1):
...
-        if (error instanceof OfficeExtension.Error) {
-            console.log("Debug info: " + JSON.stringify(error.debugInfo));
-        }
+        await context.sync();
+
+        console.log(cell.address);
     });
 Excel.Worksheet#getRange:member(1):
   - |-
     // Below example uses range address to get the range object.
```
Comment (alison-mk) [inline docs/code-snippets/excel-snippets.yaml]:
```suggestion
    // Use the range address to get the range object.
```

**Review by alison-mk — APPROVED** [review]
Approved with a flurry of suggestions to improve legacy code comments.

**Comment by AlexJerabek** [comment]
Thanks @alison-mk. I have no idea why comments on this PR are becoming immediately outdated, but I've manually added your suggestions and will rerun the scripts.

**Pull request merged**